src/metrics/misc.ts

```diff
@@ -17,3 +26,163 @@ export function countGamesInDataset(datasetPath: string): number {
+    console.log(`Player with most games played: ${this.playerMostGames}`)
+    console.log('Number of games played by time control type: '), console.table(this.gameTypeStats);
+    console.log('Number of games played by time control quantity: '), console.table(this.gameTimeControlStats);
+  }
+
+  // Aggregate the results of the metric
+  aggregate() {
+    // Calculate the average player rating after each game
+    this.averagePlayerRating = this.totalPlayerRating / (this.numberGamesAnalyzed * 2);
+
+    // Calculate the average player rating diff
+    this.averageRatingDiff = this.totalPlayerRatingDiff / this.numberGamesAnalyzed
+
+    // Calculate the player with the most games played
+    let maxGames = 0;
+    let playerMostGames = '';
+    for (const player in this.playerGameStats) {
+      if (this.playerGameStats[player] > maxGames) {
+        maxGames = this.playerGameStats[player];
+        playerMostGames = player;
+      }
+    }
+    this.mostGamesPlayed = maxGames;
+    this.playerMostGames = playerMostGames;
+
+    return {
+      averagePlayerRating: this.averagePlayerRating,
+      averageRatingDiff: this.averageRatingDiff,
+      mostGamesPlayed: maxGames,
+      playerMostGames,
+    };
+  }
+
+  // Reset the maps used to track metrics
+  clear(): void {
+    this.numberGamesAnalyzed = 0;
+    this.averagePlayerRating = 0;
+    this.averageRatingDiff = 0;
+    this.largestRatingDiff = 0;
+    this.largestRatingDiffGame = '';
+    this.mostGamesPlayed = 0;
+    this.playerMostGames = '';
+    this.gameTypeStats = {
+      numberBulletGames: 0,
+      numberBlitzGames: 0,
+      numberRapidGames: 0,
+      numberClassicalGames: 0,
+    };
+    this.gameTimeControlStats = {};
+
+    this.totalPlayerRating = 0;
+    this.totalPlayerRatingDiff = 0;
+    this.playerGameStats = {};
+
+  }
+
+  processGame(
+    game: { move: PrettyMove; board: Piece[] }[],
+    metadata?: string[]
+  ) {
+
+    // Update the gameTimeControlStats based on the time control of the game
+    const timeControl = metadata?.find((data) => data.startsWith('[TimeControl'));
+    if (timeControl) {
+      const timeControlQuantity = timeControl.split(' ')[1];
+      if (this.gameTimeControlStats[timeControlQuantity]) {
+        this.gameTimeControlStats[timeControlQuantity]++;
+      } else {
+        this.gameTimeControlStats[timeControlQuantity] = 1;
+      }
+    }
+    
+    // Identify the time control type from the metadata
+    const gameType = metadata?.find((data) => data.startsWith('[Event'));
+    if (gameType) {
+      if (gameType.includes('Bullet')) {
+        this.gameTypeStats.numberBulletGames++;
+      } else if (gameType.includes('Blitz')) {
+        this.gameTypeStats.numberBlitzGames++;
+      } else if (gameType.includes('Rapid')) {
+        this.gameTypeStats.numberRapidGames++;
+      } else if (gameType.includes('Classical')) {
+        this.gameTypeStats.numberClassicalGames++;
+      }
```

Comment:
can we use an exact match instead of the string search method?

Reply:
alternative is something like this but im worried with split that if we get an unexpected string in event (like "unrated tournament blitz game" or something like that it wouldn't work. so i'd rather search (can test speed later if need be)

const eventString = metadata?.find((data) => data.startsWith('[Event'));
if (eventString) {
  const gameType = eventString.split('"')[1].split(' ')[1];
  if (gameType === 'Bullet') {
    this.gameTypeStats.numberBulletGames++;

Reply:
wouldn't we want to throw an error if we get a new game type?

Reply:
good pt, will add in